src/main/scala/is/hail/utils/richUtils/RichIndexedRowMatrix.scala

```diff
@@ -21,140 +26,57 @@ object RichIndexedRowMatrix {
         s"rowsPerBlock needs to be greater than 0. rowsPerBlock: $rowsPerBlock")
       require(colsPerBlock > 0,
         s"colsPerBlock needs to be greater than 0. colsPerBlock: $colsPerBlock")
-      val m = indexedRowMatrix.numRows().toInt
-      val n = indexedRowMatrix.numCols().toInt
 
-      val blocksRDD: RDD[((Int, Int), Matrix)] = indexedRowMatrix.rows.flatMap({ ir =>
+      val m = indexedRowMatrix.numRows()
+      val n = indexedRowMatrix.numCols()
+      val lastRowBlockIndex = m / rowsPerBlock
+      val lastColBlockIndex = n / colsPerBlock
+      val lastRowBlockSize = (m % rowsPerBlock).toInt
+      val lastColBlockSize = (n % colsPerBlock).toInt
+      val numRowBlocks = math.ceil(m.toDouble / rowsPerBlock).toInt
+      val numColBlocks = math.ceil(n.toDouble / colsPerBlock).toInt
```

Comment:
For positive integer division that rounds towards positive infinity, I prefer this expression:

```scala
x / y + (x % y != 0).toInt
```

Ideally this can be calculated in three integer processor instructions because `div` usually returns the remainder and boolean results are already `Int`s.